### DISCUSSION TITLE
Need EJBCA CLI command for renew certificate using same key

### DISCUSSION THREAD

**Discussion by mvpnitesh** [Keyfactor/ejbca-ce#628] [General]
Hi EJBCA,

I want to renew the certificate using EJBCA CLI , I found [https://doc.primekey.com/ejbca/ejbca-operations/ejbca-operations-guide/ca-operations-guide/end-entities/certificate-renewal] which is a GUI link  for certificate renewal, but i'm looking for a ejbca CLI to renew the certificate of an end entity. I am not able to find the steps anywhere. Can you please help me?

**Comment by primetomas** [comment]
Something like this you mean?
https://doc.primekey.com/ejbca/ejbca-operations/ejbca-operations-guide/ca-operations-guide/end-entities/ssl-certificate-expiration

**Reply by mvpnitesh** [reply]
Thank you @primetomas for the response

We are using EJBCA community addition for creating Intermediate CA certs and Leaf Certs using cmpclient and EJBCA container. Upon certificate CA or Leaf certificate expiry , i'm looking for automatic renewal of those certs.
1) To create a new certificate "ejbca createcert"
2) To get the certificate details "ejbca ra getendentitycert"
3) To revoke the certificate detail "ejbca ra revokecert"

Similary i'm looking for a single command to renew the certificate.

If i follow [renew-link](https://doc.primekey.com/ejbca/ejbca-operations/ejbca-operations-guide/ca-operations-guide/end-entities/ssl-certificate-expiration) we are setting the endentity status to 10 and adding the new key and running all the steps , i'm looking for a single command which renews the cert or some command like "ejbca ca renewca"

**Reply by primetomas** [reply]
There is no such CLI command. You need to write a script, use ansible or another automation solution. You can do a lot with ansible, use our ansible repo as inspiration. I don’t know if it has CA renewal or not, perhaps it does.

**Reply by mvpnitesh** [reply]
Can you please share your ansible repo for EJBCA ?

**Reply by primetomas** [reply]
https://github.com/Keyfactor/ansible-ejbca-signserver-playbooks

You can also search “ansible” in the Keyfactor GitHub to find many different things (most not for EJBCA but still PKI related).

regards

**Reply by mvpnitesh** [reply]
Thank you @primetomas
